doc/developer/style.md

```diff
@@ -122,6 +122,16 @@ Some key principles to highlight are:
 * Avoid contractions and spell out words in full.
 * Word choices to be mindful of: find vs. exists, may vs. can vs. might.
 
+### System catalog naming
+We adhere to naming standards for our system catalog relations (tables, views), which includes both the stable `mz_catalog` relations and the unstable `mz_internal` relations.
+
+Catalog relation names should be consistent with the user-facing naming and messaging in our docs. The names should not reference internal-only concepts when possible.
+
+Singular vs Plural:
+* Relations representing objects in our system are plural. Tables and views that list "all items of type x in the system" are the typical case. Examples are `mz_sources`, `mz_connections`. This is consistent with the precedent set in the existing `mz_catalog` relations.
+* Relations representing properties of an objects, where each record has a mapping to a system objects, are singular. Examples are `mz_source_status` and `mz_source_status_history`.
+* When in doubt, use singular.
```

Comment:
I would like to contest this guidance! Historically, the rule has been "always use plural." A few relations got added that don't follow the rule, and I'm not convinced that we should be changing the guidance, instead of changing the exceptions! 

There _does_ seem to be a fundamental exception we need to make here, as it seems there is widespread dislike of naming tables `mz_storage_usages` or `mz_storage_utilizations` or `mz_storage_histories`. But I don't think the rule is about relations that describe objects vs relations that describe properties of those objects. That's a fuzzy line, too, as e.g. `mz_kafka_connections` is _both_ a relation that describes kafka connection objects, and a relation that describes properties of things in `mz_connections`.

I think the rule is something more along the lines of: if the final noun in the table is a collective noun, don't pluralize it. So `_usage` and `_history` tables don't get pluralized, because "usage" and "history" are collective. I think "usage" and "history" might be an exhaustive list of the nouns that will appear in the system catalog that don't get pluralized. This is, AFAICT, consistent with how Snowflake's system catalog is named, which is a nice plus: https://docs.snowflake.com/en/sql-reference/account-usage.html#account-usage-views.

IMO, though, it should be `mz_source_statuses` and not `mz_source_status`, and likewise for `mz_sink_statuses`.

Reply:
^ This is more intuitive and is a better articulation of what we were trying to achieve with `[thing]_statuses` and `[thing]_utilization`. To go one step further, if we tack on another noun do we get both `[thing]_statuses` and `[thing]_status_history`? I think yes.

Reply:
Works for me.

For completeness, there are a lot of compelling arguments for being all singular too, and that's what postgres does for its system tables https://www.postgresql.org/docs/current/catalogs.html, but given we already have a lot of relations that are plural, it's lowest friction to stick with the historical rule.

Reply:
> But, weirdly, not its system views. Almost all of the system views are plural: https://www.postgresql.org/docs/current/views.html. It's weird. The SQL standard information_schema (https://www.postgresql.org/docs/current/information-schema.html) uses the plural, too.

Hah - moral of the story...it's complicated and at least we'll be more consistent than postgres.

I'll go ahead and update this PR with the conclusions from this thread barring any strong objections.